platform/ios/ios.xcodeproj/project.pbxproj

```diff
@@ -2762,15 +2768,21 @@
 				CLANG_CXX_LIBRARY = "libc++";
 				CLANG_ENABLE_MODULES = YES;
 				CLANG_ENABLE_OBJC_ARC = YES;
+				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
 				CLANG_WARN_BOOL_CONVERSION = YES;
+				CLANG_WARN_COMMA = YES;
 				CLANG_WARN_CONSTANT_CONVERSION = YES;
 				CLANG_WARN_DIRECT_OBJC_ISA_USAGE = YES_ERROR;
 				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
 				CLANG_WARN_EMPTY_BODY = YES;
 				CLANG_WARN_ENUM_CONVERSION = YES;
 				CLANG_WARN_INFINITE_RECURSION = YES;
 				CLANG_WARN_INT_CONVERSION = YES;
+				CLANG_WARN_NON_LITERAL_NULL_CONVERSION = YES;
+				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
+				CLANG_WARN_STRICT_PROTOTYPES = YES;
```

Comment:
We're also maintaining a list of these in https://github.com/mapbox/mapbox-gl-native/blob/f16ecf59074e6c1f6296eb03f30b744eaa66779f/cmake/mbgl.cmake#L127-L151 so that they're aligned across all targets.

Reply:
This macro isn’t currently used with `ios.xcodeproj` — these settings are left at their defaults for that non-generated project, some of which are different than what we define in this macro.

The changes in this PR are new defaults in Xcode 9 that we’re accepting in `ios.xcodeproj`. We can also choose to accept these defaults for the generated `mbgl.xcodeproj`, but that’s something I haven’t done yet.

Here are the new defaults that Xcode thinks we should enable for `mbgl.xcodeproj`:

<img width="419" alt="screen shot 2017-12-05 at 11 22 41 am" src="https://user-images.githubusercontent.com/1198851/33617877-f43a4d1c-d9ae-11e7-9130-803e8272fc63.png">

Reply:
Looks like we’re covered on most of them already. I’ve added the relevant new suggested warnings in f4021c4.